src/pbf/mod.rs

```diff
@@ -136,16 +137,27 @@ fn decode_dense_nodes(
         // last_* start off 0
         let id = ids[index] + last_id;
         last_id = id;
-        let lat = lats[index] + last_lat;
-        last_lat = lat;
-        let lon = lons[index] + last_lon;
-        last_lon = lon;
 
-        // FIXME lat/lon here seem to be missing the last digit compared to the website
-        let lat = lat_offset + (granularity * lat);
-        let lat = 0.000000001 * (lat as f32);
-        let lon = lon_offset + (granularity * lon);
-        let lon = 0.000000001 * (lon as f32);
+        let raw_lat = i32::try_from(lats[index] + last_raw_lat as i64)
+            .expect("raw_lat was larger than the OSM precision allows");
+        last_raw_lat = raw_lat;
+
+        let raw_lon = i32::try_from(lons[index] + last_raw_lon as i64)
+            .expect("raw_lon was larger than OSM precision allows");
+        last_raw_lon = raw_lon;
+
+        // granularity is in nanodegrees
+        let scale_factor = granularity / COORD_PRECISION_NANOS;
+        let mut internal_lat = raw_lat * scale_factor;
+        let mut internal_lon = raw_lon * scale_factor;
+
+        // Offsets from pbf are in nanodegrees
+        let internal_lat_offset = lat_offset / COORD_PRECISION_NANOS as i64;
```

Comment:
There were some non-trivial changes here. I confess that I didn't test anything other than the default offset (`0`) and default granularity (`100`)